### PR TITLE
Upgrade to Micrometer 1.6.1

### LINKED ISSUE #13079: Upgrade Micrometer to 1.6.1
Need to work around a few different issues before moving up to Micrometer 1.6.0 

* Some incubating API methods were dropped in 1.6.0 (pre-register tags on Timer Sample)
* micrometer-metrics/micrometer#2330 -- NullPointerException in JvmMemory.isOldGenPool in 1.6.0 for AdoptOpenJDK
* micrometer-metrics/micrometer#2334 -- IllegalArgumentException for Hibernate

### PR DESCRIPTION
**Pull request by ebullient**
Address API changes (dropped Incubator methods)
Consolidate to one mechanism for registering metrics, w/ aligned names / labels

Resolves #13079

### CODE REVIEW

**Comment by ebullient** [comment]
still working through builds, etc. so nothing is surprising (I should have pushed the draft button.. oops)

File: extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/metrics/HibernateMetricsRecorder.java [hunk 1]
```diff
@@ -58,118 +45,224 @@ public void accept(MetricsFactory metricsFactory) {
...
      */
     void registerMetrics(MetricsFactory metricsFactory, String puName, Statistics statistics) {
-        createStatisticsCounter(metricsFactory, "hibernate-orm.sessions.open",
+
+        // Session statistics
+        createStatisticsCounter(metricsFactory, "hibernate.sessions.open",
                 "Global number of sessions opened",
                 puName, statistics, Statistics::getSessionOpenCount);
-        createStatisticsCounter(metricsFactory, "hibernate-orm.sessions.closed",
+        createStatisticsCounter(metricsFactory, "hibernate.sessions.closed",
```
Comment (Sanne) [inline extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/metrics/HibernateMetricsRecorder.java:53]: :+1:

File: extensions/micrometer/runtime/src/main/java/io/micrometer/core/instrument/binder/jvm/JvmMemory.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        if (usage == null) {
+            return Double.NaN;
+        }
+        return getter.applyAsLong(usage);
+    }
+
+    @Nullable
+    private static MemoryUsage getUsage(MemoryPoolMXBean memoryPoolMXBean) {
+        try {
+            return memoryPoolMXBean.getUsage();
```
Comment (Sanne) [inline extensions/micrometer/runtime/src/main/java/io/micrometer/core/instrument/binder/jvm/JvmMemory.java:76]: How is this going to be handled in native mode?

Comment (ebullient) [reply extensions/micrometer/runtime/src/main/java/io/micrometer/core/instrument/binder/jvm/JvmMemory.java:76]: The usage of this method is guarded.. but you've reminded me that I have to work around another 1.6.0 issue..

**Comment by ebullient** [comment]
I'm going to pause on this. 1.6.1 should be coming soon with some patches that I'd rather wait for than work around

File: extensions/micrometer/runtime/src/main/java/io/micrometer/core/instrument/binder/jvm/JvmMemory.java (hunk 2 above)
Comment (ebullient) [reply extensions/micrometer/runtime/src/main/java/io/micrometer/core/instrument/binder/jvm/JvmMemory.java:76]: This is one of the reasons I want to wait for 1.6.1. Picking this up correctly was unpredictable.

**Comment by Sanne** [comment]
thanks @ebullient , please re-assign to me when the new PR is ready.

**Pull request closed without merging**